backend/ethereum/channel/funder.go

```diff
@@ -184,7 +184,7 @@ func (f *Funder) sendFundingTx(ctx context.Context, request channel.FundingReq,
 	// nolint: gocritic
 	if bal == nil || bal.Sign() <= 0 {
 		f.log.WithFields(log.Fields{"channel": request.Params.ID(), "idx": request.Idx}).Debug("Skipped zero funding.")
-	} else if alreadyFunded, err := checkFunded(ctx, bal, contract, fundingID); err != nil {
+	} else if alreadyFunded, err := f.checkFunded(ctx, bal, contract, fundingID); err != nil {
```

Comment:
Not sure why we check the funding here before we actually do it, but this should probably be not the subject of the current PR.

Reply:
The framework could be restarted after sending the `deposit` transactions, in which case we do not want to send them twice.  
It is still a race condition in that case, but should mitigate most of it. This is what the docu means with `It is idempotent.`.  
@matthiasgeihs